### PR TITLE
use listmode time frame for output frame definition if all events are used

### LINKED ISSUE #1512: Lmtoprojdata image duration dpends on time frame definitions
When creating the exam info LMtoProjdata get the examinfo from the LM but then set timeframe according to  time frame definitions, so if this was not set in the parfile we will have a time duration of 0 even though the LM exam info had a value>0

https://github.com/UCL/STIR/blob/6417022e54a4814f25776bc8ca1df3acd4e865f6/src/listmode_buildblock/LmToProjData.cxx#L661-L665

or maybe I am missing something

### PR DESCRIPTION
**Pull request by danieldeidda**
use interfile information for frame difinition if all event are used,… throwing a warning otherwise

## Changes in this pull request
use interfile information for frame difinition if all event are used,… throwing a warning otherwise

## Testing performed
yes

## Related issues
--fixes #1512   

## Checklist before requesting a review

  - [v] I have performed a self-review of my code
  - [] I have added docstrings/doxygen in line with the guidance in the developer guide
  - [] I have implemented unit tests that cover any new or modified functionality (if applicable)
  - [v] The code builds and runs on my machine
  - [] `documentation/release_XXX.md` has been updated with any functionality change (if applicable)

### CODE REVIEW

File: src/listmode_buildblock/LmToProjData.cxx [hunk 1]
```diff
@@ -442,8 +442,9 @@ LmToProjData::set_up()
   else if (frame_defs.get_num_frames() < 1)
     {
       // make a single frame starting from 0. End value will be ignored.
-      vector<pair<double, double>> frame_times(1, pair<double, double>(0, 0));
-      frame_defs = TimeFrameDefinitions(frame_times);
+      frame_defs = lm_data_ptr->get_exam_info_sptr()->get_time_frame_definitions();
+      if (num_events_to_store!=0)
+          warning("A sub-sumple of the total events has been selected. The frame duration will be incorrect");
```
Comment (KrisThielemans) [inline src/listmode_buildblock/LmToProjData.cxx:447]:
```suggestion
          warning("LmToProjData: num_events_to_store has been selected. The frame duration in the Interfile header will be incorrect!");
```

**Comment by KrisThielemans** [comment]
most jobs were ok, but one
https://github.com/UCL/STIR/pull/1519/checks#step:17:1059
can you check the log file (downloadable from the Actions page).

Also, clang-format issues.

**Comment by KrisThielemans** [comment]
Can be reproduced by using the following .par file
```
lm to projdata parameters :=
input file := root_header.hroot
template projdata := template_for_ROOT_scanner.hs
frame definition file :=
num events to store := 0
output filename prefix := my_sinogram_counts_TOF

maximum absolute segment number to process := 3
do pre normalisation := 0
num tof bins in memory := 5
num segments in memory := 7
store prompts := 1
store delayeds := 0
list event coordinates := 0
end :=
```
and
```
INPUT_ROOT_FILE=test_PET_GATE.root lm_to_projdata my_lm_to_projdata_counts_TOF.par
```
which results in
```
Last stored event was recorded before time-tick at 0 secs
Total number of counts (either prompts/trues/delayeds) stored: 0
```
This is probably due to the fact that the `.hroot` file does not contain any time info, so we might need to check if the time from th elm file is valid or not.

File: src/listmode_buildblock/LmToProjData.cxx [hunk 2]
```diff
@@ -442,8 +442,10 @@ LmToProjData::set_up()
   else if (frame_defs.get_num_frames() < 1)
     {
       // make a single frame starting from 0. End value will be ignored.
-      vector<pair<double, double>> frame_times(1, pair<double, double>(0, 0));
-      frame_defs = TimeFrameDefinitions(frame_times);
+      frame_defs = lm_data_ptr->get_exam_info_sptr()->get_time_frame_definitions();
```
Comment (KrisThielemans) [inline src/listmode_buildblock/LmToProjData.cxx:445]:
```suggestion
      frame_defs = lm_data_ptr->get_exam_info_sptr()->get_time_frame_definitions();
      if (frame_defs.get_num_time_frames() == 0)
        {
          vector<pair<double, double>> frame_times(1, pair<double, double>(0, 0));
          frame_defs = TimeFrameDefinitions(frame_times);
        }
```

File: documentation/release_6.3.htm [hunk 3]
```diff
@@ -28,11 +28,15 @@ <h3>New functionality</h3>
   <li>
     Data from GE Discovery MI systems in RDF9 should now be readable. TOF information on these scanners has also been added.
     However, projection data is currently still always returned as non-TOF (but list-mode data is read as TOF).<br>
-    <a href=https://github.com/UCL/STIR/pull/1503>PR #1503</a>
+    <a href=https://github.com/UCL/STIR/pull/1519>PR #1503</a>
```
Comment (KrisThielemans) [inline documentation/release_6.3.htm:31]: This should remain 1503

File: documentation/release_6.3.htm [hunk 4]
```diff
@@ -28,11 +28,15 @@ <h3>New functionality</h3>
...
   </li>
   <li>
     <tt>stir_timings</tt> has now an extra option to parse a par-file for a projector-pair.
   </li>
+  <li>
+    Duration in sinogram interfile/exam_info obtained from lm_to_projdata has the correct value if we unlist all the events. This is not true for Root files
+    <a href=https://github.com/UCL/STIR/pull/1503>PR #1503</a>
```
Comment (KrisThielemans) [inline documentation/release_6.3.htm:38]: This has to be 1519. Also, please use ROOT, use `<tt>lm_to_projdata</tt>` and put a `<br>` after that line.

**Review by KrisThielemans — CHANGES_REQUESTED** [review]

File: documentation/release_6.3.htm (hunk 3 above)
Comment (danieldeidda) [reply documentation/release_6.3.htm:31]: sorry I changed the wrong one

**Review by KrisThielemans — APPROVED** [review]
thanks! squash-merge ok here? (we went back-and-forth a bit).

**Comment by danieldeidda** [comment]
> thanks! squash-merge ok here? (we went back-and-forth a bit).

sure

**Pull request merged**